apps/trading/client-pages/swap/swap-form.tsx

```diff
@@ -0,0 +1,201 @@
+import { useState } from 'react';
+import type { AssetFieldsFragment } from '@vegaprotocol/assets';
+import { EmblemByAsset } from '@vegaprotocol/emblem';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  TradingInput,
+  VegaIcon,
+  VegaIconNames,
+} from '@vegaprotocol/ui-toolkit';
+import { addDecimalsFormatNumber } from '@vegaprotocol/utils';
+import { useChainId } from '@vegaprotocol/wallet-react';
+import { useT } from '../../lib/use-t';
+import classNames from 'classnames';
+
+export const AssetInput = ({
+  label,
+  amount,
+  asset,
+  balance,
+  accountAssetIds,
+  assets,
+  onAmountChange,
+  onAssetChange,
+  accountWarning = true,
+  pubKey,
+}: {
+  label: string;
+  amount: string;
+  asset?: AssetFieldsFragment;
+  balance?: string;
+  accountAssetIds?: string[];
+  assets?: Record<string, AssetFieldsFragment>;
+  onAmountChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
```

Comment:
The codebase doesn't use the top level react object. Can you `import type { ChangeEvent } from 'react'`

Reply:
of course, updated.